### DISCUSSION TITLE
Where to find the keys to restore on chain wallets we create to receive BTC on RTL?

### DISCUSSION THREAD

**Discussion by SoulReaver-SS** [Ride-The-Lightning/RTL#1412] [Q&A] [answered]
I see we can keep creating wallets, but I don't see a way to get the keys for the wallets so we have the ownership of the wallets and we can restore and stuff?

**Comment by saubyk** [comment]
Hi @SoulReaver-SS this is really dependent on the underlying lightning implementation.
LND - You need to record the keys when initiating for the first time. If you don't record, there is no api available to retrieve the keys again
Core Lightning - CLN does not provide any 24 word key phrase at initialization, but you can retrieve the keys for backup. Relevant discussion here: https://github.com/ElementsProject/cln-application/discussions/34
Eclair - For onchain wallet operation it's dependent on the Bitcoin core backend

**Reply by SoulReaver-SS** [reply]
Thnx for the answer.
I'm using BTCPay Server's own LND, trying to use it w/ RDT.
I selected LND from BTCYPay, get to RDT UI, press "on chain" from left panel, go to "On-chain Transactions" panel, select address type taproot, press generate address, it gives me the new taproot address, doesn't present the corresponding key for the address. So I can't say I've seen any keys in this flow, nor during selecting Built in LND from BTCPay server.

**Reply by saubyk** [reply, marked as answer]
So, the first thing that you should understand is that, every time you generate a new address, you'll not get the "corresponding key" for that address each time. That's not how address generation works. There is only one key for the wallet and addresses are **derived** from that key. A single key can be used to derive countless addresses.

So for recovery purposes, you only need to know that one wallet key, which you should keep a backup of.

Regarding the LND key for BTCPayserver which you need to backup, you can find it on your BTCPayserver panel:
**Server Settings > Services > LND Seed Backup**
